Makefile

```diff
@@ -107,194 +107,268 @@ BASE_LDFLAGS = ${SHRINKFLAGS} \
+$(GO_MD2MAN):
+	$(call go-build,./vendor/github.com/cpuguy83/go-md2man)
 
-shellfiles: ${SHFMT}
-	$(eval SHELLFILES=$(shell ${SHFMT} -f . | grep -v vendor/ | grep -v hack/lib | grep -v hack/build-rpms.sh | grep -v .bats))
+$(GINKGO):
+	$(call go-build,./vendor/github.com/onsi/ginkgo/v2/ginkgo)
 
-shfmt: shellfiles
-	${SHFMT} -ln bash -w -i 4 -d ${SHELLFILES}
-	${SHFMT} -ln bats -w -sr -d $(BATS_FILES)
+$(RELEASE_NOTES): $(BUILD_BIN_PATH)
+	$(call curl_to,https://storage.googleapis.com/k8s-artifacts-sig-release/kubernetes/release/$(RELEASE_NOTES_VERSION)/release-notes-amd64-linux,$(RELEASE_NOTES))
 
-shellcheck: shellfiles ${SHELLCHECK}
-	${SHELLCHECK} \
-		-P scripts \
-		-P test \
-		-x \
-		${SHELLFILES} ${BATS_FILES}
+$(SHFMT): $(BUILD_BIN_PATH)
+	$(call curl_to,https://github.com/mvdan/sh/releases/download/$(SHFMT_VERSION)/shfmt_$(SHFMT_VERSION)_linux_amd64,$(SHFMT))
 
-check-nri-bats-tests: test/nri/nri.test
-	./hack/check-nri-bats-tests.sh
+$(ZEITGEIST): $(BUILD_BIN_PATH)
+	$(call curl_to,https://storage.googleapis.com/k8s-artifacts-sig-release/kubernetes-sigs/zeitgeist/$(ZEITGEIST_VERSION)/zeitgeist-amd64-linux,$(ZEITGEIST))
+
+$(MOCKGEN): $(BUILD_BIN_PATH)
+	$(call curl_to,https://github.com/golang/mock/releases/download/v$(MOCKGEN_VERSION)/mock_$(MOCKGEN_VERSION)_linux_$(GO_ARCH).tar.gz,$(BUILD_BIN_PATH)/mockgen.tar.gz)
+	tar xf $(BUILD_BIN_PATH)/mockgen.tar.gz --strip-components=1 -C $(BUILD_BIN_PATH)
+
+$(GO_MOD_OUTDATED): $(BUILD_BIN_PATH)
+	$(call curl_to,https://github.com/psampaz/go-mod-outdated/releases/download/v$(GO_MOD_OUTDATED_VERSION)/go-mod-outdated_$(GO_MOD_OUTDATED_VERSION)_Linux_x86_64.tar.gz,$(BUILD_BIN_PATH)/gmo.tar.gz)
+	tar xf $(BUILD_BIN_PATH)/gmo.tar.gz -C $(BUILD_BIN_PATH)
+
+$(GOSEC): $(BUILD_BIN_PATH)
+	$(call curl_to,https://github.com/securego/gosec/releases/download/v$(GOSEC_VERSION)/gosec_$(GOSEC_VERSION)_linux_amd64.tar.gz,$(BUILD_BIN_PATH)/gosec.tar.gz)
+	tar xf $(BUILD_BIN_PATH)/gosec.tar.gz -C $(BUILD_BIN_PATH)
+
+$(MDTOC): $(BUILD_BIN_PATH)
+	$(call curl_to,https://storage.googleapis.com/k8s-artifacts-sig-release/kubernetes-sigs/mdtoc/$(MDTOC_VERSION)/mdtoc-amd64-linux,$(MDTOC))
+
+$(GOLANGCI_LINT):
+	export VERSION=$(GOLANGCI_LINT_VERSION) \
+		URL=https://raw.githubusercontent.com/golangci/golangci-lint \
+		BINDIR=${BUILD_BIN_PATH} && \
+	curl -sSfL $$URL/$$VERSION/install.sh | sh -s $$VERSION
+
+$(SHELLCHECK): $(BUILD_BIN_PATH)
+	URL=https://github.com/koalaman/shellcheck/releases/download/$(SHELLCHECK_VERSION)/shellcheck-$(SHELLCHECK_VERSION).linux.x86_64.tar.xz \
+	SHA256SUM=f35ae15a4677945428bdfe61ccc297490d89dd1e544cc06317102637638c6deb && \
+	curl -sSfL $$URL | tar xfJ - -C ${BUILD_BIN_PATH} --strip 1 shellcheck-$(SHELLCHECK_VERSION)/shellcheck && \
+	sha256sum ${SHELLCHECK} | grep -q $$SHA256SUM
+
+
+##@ Build targets:
+
+.PHONY: binaries
+binaries: bin/crio bin/pinns ## Build all binaries.
+
+.PHONY: test-binaries
+test-binaries: ## Build all test-binaries
+test-binaries: \
```

Comment:
why are there two of these?

Reply:
Because of the backslash. This would be invalid syntax

```
test-binaries: \ ## Build all test-binaries
	test/copyimg/copyimg \
	test/checkseccomp/checkseccomp \
	test/checkcriu/checkcriu \
	test/nri/nri.test
```

We could get everything on one line but I thought having two lines instead seems fine.